packages/desktop-client/src/components/reports/graphs/tableGraph/ReportTableList.tsx

```diff
@@ -1,18 +1,79 @@
-// @ts-strict-ignore
-import React from 'react';
+import React, { type ReactNode } from 'react';
 
-import { type GroupedEntity } from 'loot-core/src/types/models/reports';
+import {
+  type GroupedEntity,
+  type DataEntity,
+} from 'loot-core/src/types/models/reports';
 
 import { type CSSProperties, theme } from '../../../../style';
 import { View } from '../../../common/View';
 import { Row } from '../../../table';
 
+type RenderRowProps = {
+  index: number;
+  parent_index?: number;
+  compact: boolean;
+  renderItem: (arg: {
+    item: GroupedEntity;
+    mode: string;
+    intervalsCount: number;
+    compact: boolean;
+    style?: CSSProperties;
+    compactStyle?: CSSProperties;
+  }) => ReactNode;
+  intervalsCount: number;
+  mode: string;
+  metadata: GroupedEntity[];
+  style?: CSSProperties;
+  compactStyle?: CSSProperties;
+};
+
+function RenderRow({
+  index,
+  parent_index,
+  compact,
+  renderItem,
+  intervalsCount,
+  mode,
+  metadata,
+  style,
+  compactStyle,
+}: RenderRowProps) {
+  const child = metadata[index];
+  const parent =
+    parent_index !== undefined ? metadata[parent_index] : ({} as GroupedEntity);
+
+  const item =
+    parent_index === undefined
+      ? child
+      : (parent.categories && parent.categories[index]) ||
+        ({} as GroupedEntity);
+
+  const renderRow = renderItem({
+    item,
+    mode,
+    intervalsCount,
+    compact,
+    style,
+    compactStyle,
+  });
+
+  return <View>{renderRow}</View>;
+}
+
 type ReportTableListProps = {
-  data: GroupedEntity;
-  mode?: string;
-  intervalsCount?: number;
+  data: DataEntity;
+  mode: string;
+  intervalsCount: number;
   groupBy: string;
-  renderItem;
+  renderItem: (arg: {
```

Comment:
nitpick: Should we extract this into it's own named type. I see we are duplicating it here and in RenderRowProps currently.